lightwallet-client-lib/src/main/java/co/electriccoin/lightwallet/client/util/Disposable.kt

```diff
@@ -0,0 +1,42 @@
+package co.electriccoin.lightwallet.client.util
+
+import java.io.IOException
+
+/**
+ * A [Disposable] is a source or destination of data that can be closed/disposed.
+ * The dispose method is invoked to release resources that the object is holding (such as open files).
+ */
+interface Disposable {
+    /**
+     * Closes this stream and releases any system resources associated
+     * with it. If the stream is already closed then invoking this
+     * method has no effect.
+     *
+     * In cases where the close may fail require careful attention. It is strongly advised
+     * to relinquish the underlying resources and to internally the [Disposable] as closed, prior to throwing
+     * the [IOException].
+     *
+     * @throws IOException if an I/O error occurs
+     */
+    @Throws(IOException::class)
+    suspend fun dispose()
+}
+
+/**
+ * Executes the given [block] function on this resource and then closes it down correctly whether an exception
+ * is thrown or not.
+ *
+ * @param block a function to process this [Disposable] resource.
+ * @return the result of [block] function invoked on this resource.
```

Comment:
`use` is called on the object holding the resource, not the resource itself:
```suggestion
 * Executes the given [block] function on this object and then closes it down correctly whether an exception
 * is thrown or not.
 *
 * @param block a function to process this [Disposable] object.
 * @return the result of [block] function invoked on this object.
```
If instead you want to refer to the `Disposable` object as a "resource", then "(underlying) resources" in the `Disposable` documentation needs to be replaced with some other wording to distinguish the two.